install.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash
+# Script for installing opengrep on Linux and MacOS
+set -e
+
+# Color output
+YELLOW='\033[0;33m'
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+
+# Global variable for temporary directory
+TEMP_DIR=""
+
+# Cleanup function
+cleanup() {
+    local exit_code=$?
+    printf "%b" "\n${YELLOW}Cleaning up...${NC}\n"
+    
+    # Remove temporary directory if it exists
+    if [ -n "$TEMP_DIR" ] && [ -d "$TEMP_DIR" ]; then
+        rm -rf "$TEMP_DIR"
```

Comment:
In the event a bug is introduced into this script, this can be very dangerous. 

As mentioned in my other commend, please consider downloading directly into the install directory.